helix-term/src/commands/lsp.rs

```diff
@@ -1350,22 +1384,22 @@ pub fn rename_symbol(cx: &mut Context) {
                 if event != PromptEvent::Validate {
                     return;
                 }
-
                 let (view, doc) = current!(cx.editor);
-                let language_server = language_server!(cx.editor, doc);
-                let offset_encoding = language_server.offset_encoding();
 
+                let Some(language_server) = doc
+                    .language_servers_with_feature(LanguageServerFeature::RenameSymbol)
+                    .find(|ls| language_server_id.is_none() || Some(ls.id()) == language_server_id)
```

Comment:
```suggestion
                    .language_servers_with_feature(LanguageServerFeature::RenameSymbol)
                    .find(|ls| language_server_id.map_or(true, |id| == ls.id())
```

apart from the minor clarity thing above this is also a bit confusing at first glance, maybe add a comment: if a specific `language_server` is set we find that LS otherwise we use the first LS that supports renaming